core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -194,6 +194,13 @@ public synchronized void connect() throws IOException, ConnectionFailedException
       } catch (TTransportException e) {
         LOG.error("Failed to connect (" + retry.getRetryCount() + ") to " + getServiceName() + " "
             + mMode + " @ " + mAddress + " : " + e.getMessage());
+        if (e.getMessage().contains("java.net.SocketTimeoutException")) {
+          // Do not retry if socket timeout.
+          String message = "Thrift transport open times out. Please check whether the "
+              + "authentication types match between client and server. Note that NOSASL client "
+              + "is not able to connect to servers with SIMPLE security mode.";
+          throw new IOException(message, e);
```

Comment:
Yeah, I agree that it might make sense to separate connection timeout from socket timeout. It's ok for socket timeout to be a bit longer and actually the `transport.open` timeout here depends on the socket timeout value. @calvinjia for some input.